### PR TITLE
WFCORE-3120 - fix and reenable SlaveHostControllerElytronAuthenticationTestCase.slaveWithPlainMechanism()

### PR DESCRIPTION
**Pull request by luck3y**

### CODE REVIEW

**Comment by bstansberry** [comment]
Windows test failure is related.

**Comment by luck3y** [comment]
Yeah, odd. It seems to have just not waited for the slave. Gonna retest and see if it happens again, and look locally.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/elytron/SlaveHostControllerElytronAuthenticationTestCase.java [hunk 1]
```diff
@@ -118,13 +118,15 @@ private void slaveWithPlainMechanism() throws Exception {
...
+        testSupport.getDomainSlaveLifecycleUtil().awaitHostController(System.currentTimeMillis());
         readHostControllerStatus(getDomainMasterClient());
 
         // Set the allowed mechanism back to Digest-MD5
         getDomainMasterClient().execute(changePresentedMechanisms("master", new HashSet<>(Arrays.asList("DIGEST-MD5"))));
         getDomainSlaveClient().execute(changeSaslMechanism("slave", "DIGEST-MD5"));
         reloadSlave();
         testSupport.getDomainSlaveLifecycleUtil().awaitHostController(System.currentTimeMillis());
+        assertTrue("Host \"slave\" did not reconnect to master with plain mechanism.", hostRunning(FAILED_RELOAD_TIMEOUT_MILLIS));
         readHostControllerStatus(domainMasterClient);
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/elytron/SlaveHostControllerElytronAuthenticationTestCase.java]: L129 makes L130 redundant.

This is OT to this PR but the other use of hostRunning in this class kind of sucks. AIUI it basically means the test will wait around for 10 secs waiting for the slave to *not* reauth. We need to be more clever.

Comment (luck3y) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/elytron/SlaveHostControllerElytronAuthenticationTestCase.java]: Yeah, agreed on the redundancy, I should have removed that one earlier.

Good point on the other test, I'd been focusing on the broken one, but I'd wondered about that when I'd look at it before. I'll have a think about how we could do something better.

**Comment by luck3y** [comment]
@bstansberry WDYT about this? It tails the slave logs and waits for WFLYHC0043: from 'WFLYHC0043: Unable to connect due to authentication failure.' before verifying the slave isn't registered with the host.

Runs in 149ms in my IDE ;)

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Pull request merged**